math/pythagoras/pythagoras_test.go

```diff
@@ -0,0 +1,31 @@
+package pythagoras
+
+import (
+	"math"
+	"testing"
+)
+
+// TableDrivenTest for checking multiple values against our Test Function
+var distanceTest = []struct {
+	v1  Vector
+	v2  Vector
+	res float64
+}{
+	{Vector{2, -1, 7}, Vector{1, -3, 5}, 3.0},
+	{Vector{4, 10, 9}, Vector{4, 3, 5}, 8.06},
+	{Vector{8, 5, 5}, Vector{1, 1, 12}, 10.67},
+	{Vector{1, 1, 1}, Vector{2, 2, 2}, 1.73},
+}
+
+//TestDistance tests the Function Distance with 2 vectors
+func TestDistance(t *testing.T) {
+	for _, tt := range distanceTest {
+		res := Distance(tt.v1, tt.v2)           // Calculate result for
```

Comment:
Please enclose each iteration in a t.Run() func. This would ensure that even if one of the test cases fail, the others will still be executed. The func will require a name for the test case which you can add as a part of distanceTest slice. You may use [this test file](math/primecheck/prime_check_test.go) for  reference.